From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 21 Apr 2022 01:24:45 +0200
Subject: osk-layouts: Fix Swiss layouts

Commit fe3a80ab9fcf added layouts for both Swiss French and Swiss
French, but as neither name matches the XKB layout code, they are
not used and the OSK falls back to the US layout.

The correct names are 'ch' (for Switch German) and 'ch+fr', so
rename the layout definitions accordingly.

Bug: https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/5372
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2273>
Origin: upstream, 42.1, commit:b9b96e2b5ba470d399a314811f66889852d59dc1
---
 data/gnome-shell-osk-layouts.gresource.xml |   4 +-
 data/osk-layouts/ch+fr.json                | 570 +++++++++++++++++++++++++++++
 data/osk-layouts/ch.json                   | 570 +++++++++++++++++++++++++++++
 data/osk-layouts/de+ch.json                | 570 -----------------------------
 data/osk-layouts/fr+ch.json                | 570 -----------------------------
 5 files changed, 1142 insertions(+), 1142 deletions(-)
 create mode 100644 data/osk-layouts/ch+fr.json
 create mode 100644 data/osk-layouts/ch.json
 delete mode 100644 data/osk-layouts/de+ch.json
 delete mode 100644 data/osk-layouts/fr+ch.json

diff --git a/data/gnome-shell-osk-layouts.gresource.xml b/data/gnome-shell-osk-layouts.gresource.xml
index ad6dc99..c5f22ef 100644
--- a/data/gnome-shell-osk-layouts.gresource.xml
+++ b/data/gnome-shell-osk-layouts.gresource.xml
@@ -7,9 +7,10 @@
     <file>bg.json</file>
     <file>by.json</file>
     <file>ca.json</file>
+    <file>ch.json</file>
+    <file>ch+fr.json</file>
     <file>cz.json</file>
     <file>de.json</file>
-    <file>de+ch.json</file>
     <file>dk.json</file>
     <file>ee.json</file>
     <file>epo.json</file>
@@ -17,7 +18,6 @@
     <file>es.json</file>
     <file>fi.json</file>
     <file>fr.json</file>
-    <file>fr+ch.json</file>
     <file>ge.json</file>
     <file>gr.json</file>
     <file>hr.json</file>
diff --git a/data/osk-layouts/ch+fr.json b/data/osk-layouts/ch+fr.json
new file mode 100644
index 0000000..d8b6ea7
--- /dev/null
+++ b/data/osk-layouts/ch+fr.json
@@ -0,0 +1,570 @@
+{
+  "levels": [
+    {
+      "level": "",
+      "mode": "default",
+      "rows": [
+        [
+          [
+            "q"
+          ],
+          [
+            "w"
+          ],
+          [
+            "e",
+            "é",
+            "è",
+            "ê",
+            "ë",
+            "ė"
+          ],
+          [
+            "r"
+          ],
+          [
+            "t"
+          ],
+          [
+            "z"
+          ],
+          [
+            "u",
+            "ü",
+            "û",
+            "ù",
+            "ú",
+            "ū"
+          ],
+          [
+            "i"
+          ],
+          [
+            "o",
+            "ö",
+            "ô",
+            "ò",
+            "ó",
+            "õ",
+            "œ",
+            "ø",
+            "ō"
+          ],
+          [
+            "p"
+          ]
+        ],
+        [
+          [
+            "a",
+            "ä",
+            "â",
+            "à",
+            "á",
+            "æ",
+            "ã",
+            "å",
+            "ā"
+          ],
+          [
+            "s",
+            "ß",
+            "ś",
+            "š"
+          ],
+          [
+            "d"
+          ],
+          [
+            "f"
+          ],
+          [
+            "g"
+          ],
+          [
+            "h"
+          ],
+          [
+            "j"
+          ],
+          [
+            "k"
+          ],
+          [
+            "l"
+          ]
+        ],
+        [
+          [
+            "y"
+          ],
+          [
+            "x"
+          ],
+          [
+            "c"
+          ],
+          [
+            "v"
+          ],
+          [
+            "b"
+          ],
+          [
+            "n",
+            "ñ",
+            "ń"
+          ],
+          [
+            "m"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "shift",
+      "mode": "latched",
+      "rows": [
+        [
+          [
+            "Q"
+          ],
+          [
+            "W"
+          ],
+          [
+            "E",
+            "É",
+            "È",
+            "Ê",
+            "Ë",
+            "Ė"
+          ],
+          [
+            "R"
+          ],
+          [
+            "T"
+          ],
+          [
+            "Z"
+          ],
+          [
+            "U",
+            "Ü",
+            "Û",
+            "Ù",
+            "Ú",
+            "Ū"
+          ],
+          [
+            "I"
+          ],
+          [
+            "O",
+            "Ö",
+            "Ô",
+            "Ò",
+            "Ó",
+            "Õ",
+            "Œ",
+            "Ø",
+            "Ō"
+          ],
+          [
+            "P"
+          ]
+        ],
+        [
+          [
+            "A",
+            "Ä",
+            "Â",
+            "À",
+            "Á",
+            "Æ",
+            "Ã",
+            "Å",
+            "Ā"
+          ],
+          [
+            "S",
+            "SS",
+            "Ś",
+            "Š"
+          ],
+          [
+            "D"
+          ],
+          [
+            "F"
+          ],
+          [
+            "G"
+          ],
+          [
+            "H"
+          ],
+          [
+            "J"
+          ],
+          [
+            "K"
+          ],
+          [
+            "L"
+          ]
+        ],
+        [
+          [
+            "Y"
+          ],
+          [
+            "X"
+          ],
+          [
+            "C"
+          ],
+          [
+            "V"
+          ],
+          [
+            "B"
+          ],
+          [
+            "N",
+            "Ñ",
+            "Ń"
+          ],
+          [
+            "M"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "1",
+            "¹",
+            "½",
+            "⅓",
+            "¼",
+            "⅛"
+          ],
+          [
+            "2",
+            "²",
+            "⅔"
+          ],
+          [
+            "3",
+            "³",
+            "¾",
+            "⅜"
+          ],
+          [
+            "4",
+            "⁴"
+          ],
+          [
+            "5",
+            "⅝"
+          ],
+          [
+            "6"
+          ],
+          [
+            "7",
+            "⅞"
+          ],
+          [
+            "8"
+          ],
+          [
+            "9"
+          ],
+          [
+            "0",
+            "ⁿ",
+            "∅"
+          ]
+        ],
+        [
+          [
+            "@"
+          ],
+          [
+            "#"
+          ],
+          [
+            "€",
+            "¢",
+            "£",
+            "$",
+            "¥",
+            "₱"
+          ],
+          [
+            "%",
+            "‰"
+          ],
+          [
+            "&"
+          ],
+          [
+            "-",
+            "_",
+            "–",
+            "—",
+            "·"
+          ],
+          [
+            "+",
+            "±"
+          ],
+          [
+            "(",
+            "<",
+            "{",
+            "["
+          ],
+          [
+            ")",
+            ">",
+            "}",
+            "]"
+          ]
+        ],
+        [
+          [
+            "*",
+            "†",
+            "‡",
+            "★"
+          ],
+          [
+            "\"",
+            "“",
+            "”",
+            "«",
+            "»"
+          ],
+          [
+            "'",
+            "‘",
+            "’",
+            "‹",
+            "›"
+          ],
+          [
+            ":"
+          ],
+          [
+            ";"
+          ],
+          [
+            "!",
+            "¡"
+          ],
+          [
+            "?",
+            "¿"
+          ]
+        ],
+        [
+          [
+            "_"
+          ],
+          [
+            "/"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt+shift",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "~"
+          ],
+          [
+            "`"
+          ],
+          [
+            "|"
+          ],
+          [
+            "•",
+            "♪",
+            "♥",
+            "♠",
+            "♦",
+            "♣"
+          ],
+          [
+            "√"
+          ],
+          [
+            "Π",
+            "π"
+          ],
+          [
+            "÷"
+          ],
+          [
+            "×"
+          ],
+          [
+            "¶",
+            "§"
+          ],
+          [
+            "∆"
+          ]
+        ],
+        [
+          [
+            "£"
+          ],
+          [
+            "¥"
+          ],
+          [
+            "$",
+            "¢"
+          ],
+          [
+            "¢"
+          ],
+          [
+            "^",
+            "↑",
+            "↓",
+            "←",
+            "→"
+          ],
+          [
+            "°",
+            "′",
+            "″"
+          ],
+          [
+            "=",
+            "≠",
+            "≈",
+            "∞"
+          ],
+          [
+            "{"
+          ],
+          [
+            "}"
+          ]
+        ],
+        [
+          [
+            "\\"
+          ],
+          [
+            "©"
+          ],
+          [
+            "®"
+          ],
+          [
+            "™"
+          ],
+          [
+            "℅"
+          ],
+          [
+            "["
+          ],
+          [
+            "]"
+          ]
+        ],
+        [
+          [
+            "<",
+            "‹",
+            "≤",
+            "«"
+          ],
+          [
+            ">",
+            "›",
+            "≥",
+            "»"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    }
+  ],
+  "locale": "fr-CH",
+  "name": "Swiss French"
+}
\ No newline at end of file
diff --git a/data/osk-layouts/ch.json b/data/osk-layouts/ch.json
new file mode 100644
index 0000000..dd08c67
--- /dev/null
+++ b/data/osk-layouts/ch.json
@@ -0,0 +1,570 @@
+{
+  "levels": [
+    {
+      "level": "",
+      "mode": "default",
+      "rows": [
+        [
+          [
+            "q"
+          ],
+          [
+            "w"
+          ],
+          [
+            "e",
+            "é",
+            "è",
+            "ê",
+            "ë",
+            "ė"
+          ],
+          [
+            "r"
+          ],
+          [
+            "t"
+          ],
+          [
+            "z"
+          ],
+          [
+            "u",
+            "ü",
+            "û",
+            "ù",
+            "ú",
+            "ū"
+          ],
+          [
+            "i"
+          ],
+          [
+            "o",
+            "ö",
+            "ô",
+            "ò",
+            "ó",
+            "õ",
+            "œ",
+            "ø",
+            "ō"
+          ],
+          [
+            "p"
+          ]
+        ],
+        [
+          [
+            "a",
+            "ä",
+            "â",
+            "à",
+            "á",
+            "æ",
+            "ã",
+            "å",
+            "ā"
+          ],
+          [
+            "s",
+            "ß",
+            "ś",
+            "š"
+          ],
+          [
+            "d"
+          ],
+          [
+            "f"
+          ],
+          [
+            "g"
+          ],
+          [
+            "h"
+          ],
+          [
+            "j"
+          ],
+          [
+            "k"
+          ],
+          [
+            "l"
+          ]
+        ],
+        [
+          [
+            "y"
+          ],
+          [
+            "x"
+          ],
+          [
+            "c"
+          ],
+          [
+            "v"
+          ],
+          [
+            "b"
+          ],
+          [
+            "n",
+            "ñ",
+            "ń"
+          ],
+          [
+            "m"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "shift",
+      "mode": "latched",
+      "rows": [
+        [
+          [
+            "Q"
+          ],
+          [
+            "W"
+          ],
+          [
+            "E",
+            "É",
+            "È",
+            "Ê",
+            "Ë",
+            "Ė"
+          ],
+          [
+            "R"
+          ],
+          [
+            "T"
+          ],
+          [
+            "Z"
+          ],
+          [
+            "U",
+            "Ü",
+            "Û",
+            "Ù",
+            "Ú",
+            "Ū"
+          ],
+          [
+            "I"
+          ],
+          [
+            "O",
+            "Ö",
+            "Ô",
+            "Ò",
+            "Ó",
+            "Õ",
+            "Œ",
+            "Ø",
+            "Ō"
+          ],
+          [
+            "P"
+          ]
+        ],
+        [
+          [
+            "A",
+            "Ä",
+            "Â",
+            "À",
+            "Á",
+            "Æ",
+            "Ã",
+            "Å",
+            "Ā"
+          ],
+          [
+            "S",
+            "SS",
+            "Ś",
+            "Š"
+          ],
+          [
+            "D"
+          ],
+          [
+            "F"
+          ],
+          [
+            "G"
+          ],
+          [
+            "H"
+          ],
+          [
+            "J"
+          ],
+          [
+            "K"
+          ],
+          [
+            "L"
+          ]
+        ],
+        [
+          [
+            "Y"
+          ],
+          [
+            "X"
+          ],
+          [
+            "C"
+          ],
+          [
+            "V"
+          ],
+          [
+            "B"
+          ],
+          [
+            "N",
+            "Ñ",
+            "Ń"
+          ],
+          [
+            "M"
+          ]
+        ],
+        [
+          [
+            ","
+          ],
+          [
+            " "
+          ],
+          [
+            ".",
+            "#",
+            "!",
+            ",",
+            "?",
+            "-",
+            ":",
+            "'",
+            "@"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "1",
+            "¹",
+            "½",
+            "⅓",
+            "¼",
+            "⅛"
+          ],
+          [
+            "2",
+            "²",
+            "⅔"
+          ],
+          [
+            "3",
+            "³",
+            "¾",
+            "⅜"
+          ],
+          [
+            "4",
+            "⁴"
+          ],
+          [
+            "5",
+            "⅝"
+          ],
+          [
+            "6"
+          ],
+          [
+            "7",
+            "⅞"
+          ],
+          [
+            "8"
+          ],
+          [
+            "9"
+          ],
+          [
+            "0",
+            "ⁿ",
+            "∅"
+          ]
+        ],
+        [
+          [
+            "@"
+          ],
+          [
+            "#"
+          ],
+          [
+            "€",
+            "¢",
+            "£",
+            "$",
+            "¥",
+            "₱"
+          ],
+          [
+            "%",
+            "‰"
+          ],
+          [
+            "&"
+          ],
+          [
+            "-",
+            "_",
+            "–",
+            "—",
+            "·"
+          ],
+          [
+            "+",
+            "±"
+          ],
+          [
+            "(",
+            "<",
+            "{",
+            "["
+          ],
+          [
+            ")",
+            ">",
+            "}",
+            "]"
+          ]
+        ],
+        [
+          [
+            "*",
+            "†",
+            "‡",
+            "★"
+          ],
+          [
+            "\"",
+            "“",
+            "”",
+            "«",
+            "»"
+          ],
+          [
+            "'",
+            "‘",
+            "’",
+            "‹",
+            "›"
+          ],
+          [
+            ":"
+          ],
+          [
+            ";"
+          ],
+          [
+            "!",
+            "¡"
+          ],
+          [
+            "?",
+            "¿"
+          ]
+        ],
+        [
+          [
+            "_"
+          ],
+          [
+            "/"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    },
+    {
+      "level": "opt+shift",
+      "mode": "locked",
+      "rows": [
+        [
+          [
+            "~"
+          ],
+          [
+            "`"
+          ],
+          [
+            "|"
+          ],
+          [
+            "•",
+            "♪",
+            "♥",
+            "♠",
+            "♦",
+            "♣"
+          ],
+          [
+            "√"
+          ],
+          [
+            "Π",
+            "π"
+          ],
+          [
+            "÷"
+          ],
+          [
+            "×"
+          ],
+          [
+            "¶",
+            "§"
+          ],
+          [
+            "∆"
+          ]
+        ],
+        [
+          [
+            "£"
+          ],
+          [
+            "¥"
+          ],
+          [
+            "$",
+            "¢"
+          ],
+          [
+            "¢"
+          ],
+          [
+            "^",
+            "↑",
+            "↓",
+            "←",
+            "→"
+          ],
+          [
+            "°",
+            "′",
+            "″"
+          ],
+          [
+            "=",
+            "≠",
+            "≈",
+            "∞"
+          ],
+          [
+            "{"
+          ],
+          [
+            "}"
+          ]
+        ],
+        [
+          [
+            "\\"
+          ],
+          [
+            "©"
+          ],
+          [
+            "®"
+          ],
+          [
+            "™"
+          ],
+          [
+            "℅"
+          ],
+          [
+            "["
+          ],
+          [
+            "]"
+          ]
+        ],
+        [
+          [
+            "<",
+            "‹",
+            "≤",
+            "«"
+          ],
+          [
+            ">",
+            "›",
+            "≥",
+            "»"
+          ],
+          [
+            " "
+          ],
+          [
+            ","
+          ],
+          [
+            ".",
+            "…"
+          ]
+        ]
+      ]
+    }
+  ],
+  "locale": "de-CH",
+  "name": "Swiss German"
+}
\ No newline at end of file
diff --git a/data/osk-layouts/de+ch.json b/data/osk-layouts/de+ch.json
deleted file mode 100644
index dd08c67..0000000
--- a/data/osk-layouts/de+ch.json
+++ /dev/null
@@ -1,570 +0,0 @@
-{
-  "levels": [
-    {
-      "level": "",
-      "mode": "default",
-      "rows": [
-        [
-          [
-            "q"
-          ],
-          [
-            "w"
-          ],
-          [
-            "e",
-            "é",
-            "è",
-            "ê",
-            "ë",
-            "ė"
-          ],
-          [
-            "r"
-          ],
-          [
-            "t"
-          ],
-          [
-            "z"
-          ],
-          [
-            "u",
-            "ü",
-            "û",
-            "ù",
-            "ú",
-            "ū"
-          ],
-          [
-            "i"
-          ],
-          [
-            "o",
-            "ö",
-            "ô",
-            "ò",
-            "ó",
-            "õ",
-            "œ",
-            "ø",
-            "ō"
-          ],
-          [
-            "p"
-          ]
-        ],
-        [
-          [
-            "a",
-            "ä",
-            "â",
-            "à",
-            "á",
-            "æ",
-            "ã",
-            "å",
-            "ā"
-          ],
-          [
-            "s",
-            "ß",
-            "ś",
-            "š"
-          ],
-          [
-            "d"
-          ],
-          [
-            "f"
-          ],
-          [
-            "g"
-          ],
-          [
-            "h"
-          ],
-          [
-            "j"
-          ],
-          [
-            "k"
-          ],
-          [
-            "l"
-          ]
-        ],
-        [
-          [
-            "y"
-          ],
-          [
-            "x"
-          ],
-          [
-            "c"
-          ],
-          [
-            "v"
-          ],
-          [
-            "b"
-          ],
-          [
-            "n",
-            "ñ",
-            "ń"
-          ],
-          [
-            "m"
-          ]
-        ],
-        [
-          [
-            ","
-          ],
-          [
-            " "
-          ],
-          [
-            ".",
-            "#",
-            "!",
-            ",",
-            "?",
-            "-",
-            ":",
-            "'",
-            "@"
-          ]
-        ]
-      ]
-    },
-    {
-      "level": "shift",
-      "mode": "latched",
-      "rows": [
-        [
-          [
-            "Q"
-          ],
-          [
-            "W"
-          ],
-          [
-            "E",
-            "É",
-            "È",
-            "Ê",
-            "Ë",
-            "Ė"
-          ],
-          [
-            "R"
-          ],
-          [
-            "T"
-          ],
-          [
-            "Z"
-          ],
-          [
-            "U",
-            "Ü",
-            "Û",
-            "Ù",
-            "Ú",
-            "Ū"
-          ],
-          [
-            "I"
-          ],
-          [
-            "O",
-            "Ö",
-            "Ô",
-            "Ò",
-            "Ó",
-            "Õ",
-            "Œ",
-            "Ø",
-            "Ō"
-          ],
-          [
-            "P"
-          ]
-        ],
-        [
-          [
-            "A",
-            "Ä",
-            "Â",
-            "À",
-            "Á",
-            "Æ",
-            "Ã",
-            "Å",
-            "Ā"
-          ],
-          [
-            "S",
-            "SS",
-            "Ś",
-            "Š"
-          ],
-          [
-            "D"
-          ],
-          [
-            "F"
-          ],
-          [
-            "G"
-          ],
-          [
-            "H"
-          ],
-          [
-            "J"
-          ],
-          [
-            "K"
-          ],
-          [
-            "L"
-          ]
-        ],
-        [
-          [
-            "Y"
-          ],
-          [
-            "X"
-          ],
-          [
-            "C"
-          ],
-          [
-            "V"
-          ],
-          [
-            "B"
-          ],
-          [
-            "N",
-            "Ñ",
-            "Ń"
-          ],
-          [
-            "M"
-          ]
-        ],
-        [
-          [
-            ","
-          ],
-          [
-            " "
-          ],
-          [
-            ".",
-            "#",
-            "!",
-            ",",
-            "?",
-            "-",
-            ":",
-            "'",
-            "@"
-          ]
-        ]
-      ]
-    },
-    {
-      "level": "opt",
-      "mode": "locked",
-      "rows": [
-        [
-          [
-            "1",
-            "¹",
-            "½",
-            "⅓",
-            "¼",
-            "⅛"
-          ],
-          [
-            "2",
-            "²",
-            "⅔"
-          ],
-          [
-            "3",
-            "³",
-            "¾",
-            "⅜"
-          ],
-          [
-            "4",
-            "⁴"
-          ],
-          [
-            "5",
-            "⅝"
-          ],
-          [
-            "6"
-          ],
-          [
-            "7",
-            "⅞"
-          ],
-          [
-            "8"
-          ],
-          [
-            "9"
-          ],
-          [
-            "0",
-            "ⁿ",
-            "∅"
-          ]
-        ],
-        [
-          [
-            "@"
-          ],
-          [
-            "#"
-          ],
-          [
-            "€",
-            "¢",
-            "£",
-            "$",
-            "¥",
-            "₱"
-          ],
-          [
-            "%",
-            "‰"
-          ],
-          [
-            "&"
-          ],
-          [
-            "-",
-            "_",
-            "–",
-            "—",
-            "·"
-          ],
-          [
-            "+",
-            "±"
-          ],
-          [
-            "(",
-            "<",
-            "{",
-            "["
-          ],
-          [
-            ")",
-            ">",
-            "}",
-            "]"
-          ]
-        ],
-        [
-          [
-            "*",
-            "†",
-            "‡",
-            "★"
-          ],
-          [
-            "\"",
-            "“",
-            "”",
-            "«",
-            "»"
-          ],
-          [
-            "'",
-            "‘",
-            "’",
-            "‹",
-            "›"
-          ],
-          [
-            ":"
-          ],
-          [
-            ";"
-          ],
-          [
-            "!",
-            "¡"
-          ],
-          [
-            "?",
-            "¿"
-          ]
-        ],
-        [
-          [
-            "_"
-          ],
-          [
-            "/"
-          ],
-          [
-            " "
-          ],
-          [
-            ","
-          ],
-          [
-            ".",
-            "…"
-          ]
-        ]
-      ]
-    },
-    {
-      "level": "opt+shift",
-      "mode": "locked",
-      "rows": [
-        [
-          [
-            "~"
-          ],
-          [
-            "`"
-          ],
-          [
-            "|"
-          ],
-          [
-            "•",
-            "♪",
-            "♥",
-            "♠",
-            "♦",
-            "♣"
-          ],
-          [
-            "√"
-          ],
-          [
-            "Π",
-            "π"
-          ],
-          [
-            "÷"
-          ],
-          [
-            "×"
-          ],
-          [
-            "¶",
-            "§"
-          ],
-          [
-            "∆"
-          ]
-        ],
-        [
-          [
-            "£"
-          ],
-          [
-            "¥"
-          ],
-          [
-            "$",
-            "¢"
-          ],
-          [
-            "¢"
-          ],
-          [
-            "^",
-            "↑",
-            "↓",
-            "←",
-            "→"
-          ],
-          [
-            "°",
-            "′",
-            "″"
-          ],
-          [
-            "=",
-            "≠",
-            "≈",
-            "∞"
-          ],
-          [
-            "{"
-          ],
-          [
-            "}"
-          ]
-        ],
-        [
-          [
-            "\\"
-          ],
-          [
-            "©"
-          ],
-          [
-            "®"
-          ],
-          [
-            "™"
-          ],
-          [
-            "℅"
-          ],
-          [
-            "["
-          ],
-          [
-            "]"
-          ]
-        ],
-        [
-          [
-            "<",
-            "‹",
-            "≤",
-            "«"
-          ],
-          [
-            ">",
-            "›",
-            "≥",
-            "»"
-          ],
-          [
-            " "
-          ],
-          [
-            ","
-          ],
-          [
-            ".",
-            "…"
-          ]
-        ]
-      ]
-    }
-  ],
-  "locale": "de-CH",
-  "name": "Swiss German"
-}
\ No newline at end of file
diff --git a/data/osk-layouts/fr+ch.json b/data/osk-layouts/fr+ch.json
deleted file mode 100644
index d8b6ea7..0000000
--- a/data/osk-layouts/fr+ch.json
+++ /dev/null
@@ -1,570 +0,0 @@
-{
-  "levels": [
-    {
-      "level": "",
-      "mode": "default",
-      "rows": [
-        [
-          [
-            "q"
-          ],
-          [
-            "w"
-          ],
-          [
-            "e",
-            "é",
-            "è",
-            "ê",
-            "ë",
-            "ė"
-          ],
-          [
-            "r"
-          ],
-          [
-            "t"
-          ],
-          [
-            "z"
-          ],
-          [
-            "u",
-            "ü",
-            "û",
-            "ù",
-            "ú",
-            "ū"
-          ],
-          [
-            "i"
-          ],
-          [
-            "o",
-            "ö",
-            "ô",
-            "ò",
-            "ó",
-            "õ",
-            "œ",
-            "ø",
-            "ō"
-          ],
-          [
-            "p"
-          ]
-        ],
-        [
-          [
-            "a",
-            "ä",
-            "â",
-            "à",
-            "á",
-            "æ",
-            "ã",
-            "å",
-            "ā"
-          ],
-          [
-            "s",
-            "ß",
-            "ś",
-            "š"
-          ],
-          [
-            "d"
-          ],
-          [
-            "f"
-          ],
-          [
-            "g"
-          ],
-          [
-            "h"
-          ],
-          [
-            "j"
-          ],
-          [
-            "k"
-          ],
-          [
-            "l"
-          ]
-        ],
-        [
-          [
-            "y"
-          ],
-          [
-            "x"
-          ],
-          [
-            "c"
-          ],
-          [
-            "v"
-          ],
-          [
-            "b"
-          ],
-          [
-            "n",
-            "ñ",
-            "ń"
-          ],
-          [
-            "m"
-          ]
-        ],
-        [
-          [
-            ","
-          ],
-          [
-            " "
-          ],
-          [
-            ".",
-            "#",
-            "!",
-            ",",
-            "?",
-            "-",
-            ":",
-            "'",
-            "@"
-          ]
-        ]
-      ]
-    },
-    {
-      "level": "shift",
-      "mode": "latched",
-      "rows": [
-        [
-          [
-            "Q"
-          ],
-          [
-            "W"
-          ],
-          [
-            "E",
-            "É",
-            "È",
-            "Ê",
-            "Ë",
-            "Ė"
-          ],
-          [
-            "R"
-          ],
-          [
-            "T"
-          ],
-          [
-            "Z"
-          ],
-          [
-            "U",
-            "Ü",
-            "Û",
-            "Ù",
-            "Ú",
-            "Ū"
-          ],
-          [
-            "I"
-          ],
-          [
-            "O",
-            "Ö",
-            "Ô",
-            "Ò",
-            "Ó",
-            "Õ",
-            "Œ",
-            "Ø",
-            "Ō"
-          ],
-          [
-            "P"
-          ]
-        ],
-        [
-          [
-            "A",
-            "Ä",
-            "Â",
-            "À",
-            "Á",
-            "Æ",
-            "Ã",
-            "Å",
-            "Ā"
-          ],
-          [
-            "S",
-            "SS",
-            "Ś",
-            "Š"
-          ],
-          [
-            "D"
-          ],
-          [
-            "F"
-          ],
-          [
-            "G"
-          ],
-          [
-            "H"
-          ],
-          [
-            "J"
-          ],
-          [
-            "K"
-          ],
-          [
-            "L"
-          ]
-        ],
-        [
-          [
-            "Y"
-          ],
-          [
-            "X"
-          ],
-          [
-            "C"
-          ],
-          [
-            "V"
-          ],
-          [
-            "B"
-          ],
-          [
-            "N",
-            "Ñ",
-            "Ń"
-          ],
-          [
-            "M"
-          ]
-        ],
-        [
-          [
-            ","
-          ],
-          [
-            " "
-          ],
-          [
-            ".",
-            "#",
-            "!",
-            ",",
-            "?",
-            "-",
-            ":",
-            "'",
-            "@"
-          ]
-        ]
-      ]
-    },
-    {
-      "level": "opt",
-      "mode": "locked",
-      "rows": [
-        [
-          [
-            "1",
-            "¹",
-            "½",
-            "⅓",
-            "¼",
-            "⅛"
-          ],
-          [
-            "2",
-            "²",
-            "⅔"
-          ],
-          [
-            "3",
-            "³",
-            "¾",
-            "⅜"
-          ],
-          [
-            "4",
-            "⁴"
-          ],
-          [
-            "5",
-            "⅝"
-          ],
-          [
-            "6"
-          ],
-          [
-            "7",
-            "⅞"
-          ],
-          [
-            "8"
-          ],
-          [
-            "9"
-          ],
-          [
-            "0",
-            "ⁿ",
-            "∅"
-          ]
-        ],
-        [
-          [
-            "@"
-          ],
-          [
-            "#"
-          ],
-          [
-            "€",
-            "¢",
-            "£",
-            "$",
-            "¥",
-            "₱"
-          ],
-          [
-            "%",
-            "‰"
-          ],
-          [
-            "&"
-          ],
-          [
-            "-",
-            "_",
-            "–",
-            "—",
-            "·"
-          ],
-          [
-            "+",
-            "±"
-          ],
-          [
-            "(",
-            "<",
-            "{",
-            "["
-          ],
-          [
-            ")",
-            ">",
-            "}",
-            "]"
-          ]
-        ],
-        [
-          [
-            "*",
-            "†",
-            "‡",
-            "★"
-          ],
-          [
-            "\"",
-            "“",
-            "”",
-            "«",
-            "»"
-          ],
-          [
-            "'",
-            "‘",
-            "’",
-            "‹",
-            "›"
-          ],
-          [
-            ":"
-          ],
-          [
-            ";"
-          ],
-          [
-            "!",
-            "¡"
-          ],
-          [
-            "?",
-            "¿"
-          ]
-        ],
-        [
-          [
-            "_"
-          ],
-          [
-            "/"
-          ],
-          [
-            " "
-          ],
-          [
-            ","
-          ],
-          [
-            ".",
-            "…"
-          ]
-        ]
-      ]
-    },
-    {
-      "level": "opt+shift",
-      "mode": "locked",
-      "rows": [
-        [
-          [
-            "~"
-          ],
-          [
-            "`"
-          ],
-          [
-            "|"
-          ],
-          [
-            "•",
-            "♪",
-            "♥",
-            "♠",
-            "♦",
-            "♣"
-          ],
-          [
-            "√"
-          ],
-          [
-            "Π",
-            "π"
-          ],
-          [
-            "÷"
-          ],
-          [
-            "×"
-          ],
-          [
-            "¶",
-            "§"
-          ],
-          [
-            "∆"
-          ]
-        ],
-        [
-          [
-            "£"
-          ],
-          [
-            "¥"
-          ],
-          [
-            "$",
-            "¢"
-          ],
-          [
-            "¢"
-          ],
-          [
-            "^",
-            "↑",
-            "↓",
-            "←",
-            "→"
-          ],
-          [
-            "°",
-            "′",
-            "″"
-          ],
-          [
-            "=",
-            "≠",
-            "≈",
-            "∞"
-          ],
-          [
-            "{"
-          ],
-          [
-            "}"
-          ]
-        ],
-        [
-          [
-            "\\"
-          ],
-          [
-            "©"
-          ],
-          [
-            "®"
-          ],
-          [
-            "™"
-          ],
-          [
-            "℅"
-          ],
-          [
-            "["
-          ],
-          [
-            "]"
-          ]
-        ],
-        [
-          [
-            "<",
-            "‹",
-            "≤",
-            "«"
-          ],
-          [
-            ">",
-            "›",
-            "≥",
-            "»"
-          ],
-          [
-            " "
-          ],
-          [
-            ","
-          ],
-          [
-            ".",
-            "…"
-          ]
-        ]
-      ]
-    }
-  ],
-  "locale": "fr-CH",
-  "name": "Swiss French"
-}
\ No newline at end of file
